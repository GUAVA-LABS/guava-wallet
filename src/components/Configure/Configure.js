/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { Collapse, Form, Input, Modal, Spin, Alert } from 'antd';
import {
    PlusSquareOutlined,
    WalletFilled,
    ImportOutlined,
    LockOutlined,
} from '@ant-design/icons';
import { WalletContext } from '@utils/context';
import { StyledCollapse } from '@components/Common/StyledCollapse';
import PrimaryButton, {
    SecondaryButton,
    SmartButton,
} from '@components/Common/PrimaryButton';
import {
    CashLoader,
    CashLoadingIcon,
    ThemedCopyOutlined,
    ThemedWalletOutlined,
} from '@components/Common/CustomIcons';
import { ReactComponent as Trashcan } from '@assets/trashcan.svg';
import { ReactComponent as Edit } from '@assets/edit.svg';
import { Event } from '@utils/GoogleAnalytics';

const { Panel } = Collapse;

const SettingsLink = styled.a`
    text-decoration: underline;
    color: #ff8d00;
    :visited {
        text-decoration: underline;
        color: #ff8d00;
    }
`;

const SWRow = styled.div`
    border-radius: 3px;
    padding: 10px 0;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 6px;
    @media (max-width: 500px) {
        flex-direction: column;
        margin-bottom: 12px;
    }
`;

const SWName = styled.div`
    width: 50%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    word-wrap: break-word;
    hyphens: auto;

    @media (max-width: 500px) {
        width: 100%;
        justify-content: center;
        margin-bottom: 15px;
    }

    h3 {
        font-size: 16px;
        color: #444;
        margin: 0;
        text-align: left;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
`;

const SWButtonCtn = styled.div`
    width: 50%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    @media (max-width: 500px) {
        width: 100%;
        justify-content: center;
    }

    button {
        cursor: pointer;

        @media (max-width: 768px) {
            font-size: 14px;
        }
    }

    svg {
        stroke: #444;
        fill: #444;
        width: 25px;
        height: 25px;
        margin-right: 20px;
        cursor: pointer;

        :first-child:hover {
            stroke: #ff8d00;
            fill: #ff8d00;
        }
        :hover {
            stroke: red;
            fill: red;
        }
    }
`;

const AWRow = styled.div`
    padding: 10px 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-bottom: 6px;
    h3 {
        font-size: 16px;
        display: inline-block;
        color: #444;
        margin: 0;
        text-align: left;
        font-weight: bold;
        @media (max-width: 500px) {
            font-size: 14px;
        }
    }
    h4 {
        font-size: 16px;
        display: inline-block;
        color: #ff8d00 !important;
        margin: 0;
        text-align: right;
    }
    @media (max-width: 500px) {
        flex-direction: column;
        margin-bottom: 12px;
    }
`;

const StyledConfigure = styled.div`
    h2 {
        color: #444;
        font-size: 25px;
    }
    p {
        color: #444;
    }
`;

const StyledSpacer = styled.div`
    height: 1px;
    width: 100%;
    background-color: #e2e2e2;
    margin: 60px 0 50px;
`;

const Configure = () => {
    const ContextValue = React.useContext(WalletContext);
    const { wallet, loading, apiError } = ContextValue;

    const {
        addNewSavedWallet,
        activateWallet,
        renameWallet,
        deleteWallet,
        validateMnemonic,
        getSavedWallets,
    } = ContextValue;
    const [savedWallets, setSavedWallets] = useState([]);
    const [formData, setFormData] = useState({
        dirty: true,
        mnemonic: '',
    });
    const [showRenameWalletModal, setShowRenameWalletModal] = useState(false);
    const [showDeleteWalletModal, setShowDeleteWalletModal] = useState(false);
    const [walletToBeRenamed, setWalletToBeRenamed] = useState(null);
    const [walletToBeDeleted, setWalletToBeDeleted] = useState(null);
    const [newWalletName, setNewWalletName] = useState('');
    const [
        confirmationOfWalletToBeDeleted,
        setConfirmationOfWalletToBeDeleted,
    ] = useState('');
    const [newWalletNameIsValid, setNewWalletNameIsValid] = useState(null);
    const [walletDeleteValid, setWalletDeleteValid] = useState(null);
    const [seedInput, openSeedInput] = useState(false);

    const showPopulatedDeleteWalletModal = walletInfo => {
        setWalletToBeDeleted(walletInfo);
        setShowDeleteWalletModal(true);
    };

    const showPopulatedRenameWalletModal = walletInfo => {
        setWalletToBeRenamed(walletInfo);
        setShowRenameWalletModal(true);
    };
    const cancelRenameWallet = () => {
        // Delete form value
        setNewWalletName('');
        setShowRenameWalletModal(false);
    };
    const cancelDeleteWallet = () => {
        setWalletToBeDeleted(null);
        setConfirmationOfWalletToBeDeleted('');
        setShowDeleteWalletModal(false);
    };
    const updateSavedWallets = async activeWallet => {
        if (activeWallet) {
            let savedWallets;
            try {
                savedWallets = await getSavedWallets(activeWallet);
                setSavedWallets(savedWallets);
            } catch (err) {
                console.log(`Error in getSavedWallets()`);
                console.log(err);
            }
        }
    };

    const [isValidMnemonic, setIsValidMnemonic] = useState(false);

    useEffect(() => {
        // Update savedWallets every time the active wallet changes
        updateSavedWallets(wallet);
    }, [wallet]);

    // Need this function to ensure that savedWallets are updated on new wallet creation
    const updateSavedWalletsOnCreate = async importMnemonic => {
        // Event("Category", "Action", "Label")
        // Track number of times a different wallet is activated
        Event('Configure.js', 'Create Wallet', 'New');
        const walletAdded = await addNewSavedWallet(importMnemonic);
        if (!walletAdded) {
            Modal.error({
                title: 'This wallet already exists!',
                content: 'Wallet not added',
            });
        } else {
            Modal.success({
                content: 'Wallet added to your saved wallets',
            });
        }
        await updateSavedWallets(wallet);
    };
    // Same here
    // TODO you need to lock UI here until this is complete
    // Otherwise user may try to load an already-loading wallet, wreak havoc with indexedDB
    const updateSavedWalletsOnLoad = async walletToActivate => {
        // Event("Category", "Action", "Label")
        // Track number of times a different wallet is activated
        Event('Configure.js', 'Activate', '');
        await activateWallet(walletToActivate);
        await updateSavedWallets(wallet);
    };

    async function submit() {
        setFormData({
            ...formData,
            dirty: false,
        });

        // Exit if no user input
        if (!formData.mnemonic) {
            return;
        }

        // Exit if mnemonic is invalid
        if (!isValidMnemonic) {
            return;
        }
        // Event("Category", "Action", "Label")
        // Track number of times a different wallet is activated
        Event('Configure.js', 'Create Wallet', 'Imported');
        updateSavedWalletsOnCreate(formData.mnemonic);
    }

    const handleChange = e => {
        const { value, name } = e.target;

        // Validate mnemonic on change
        // Import button should be disabled unless mnemonic is valid
        setIsValidMnemonic(validateMnemonic(value));

        setFormData(p => ({ ...p, [name]: value }));
    };

    const changeWalletName = async () => {
        if (newWalletName === '' || newWalletName.length > 24) {
            setNewWalletNameIsValid(false);
            return;
        }
        // Hide modal
        setShowRenameWalletModal(false);
        // Change wallet name
        console.log(
            `Changing wallet ${walletToBeRenamed.name} name to ${newWalletName}`,
        );
        const renameSuccess = await renameWallet(
            walletToBeRenamed.name,
            newWalletName,
        );

        if (renameSuccess) {
            Modal.success({
                content: `Wallet "${walletToBeRenamed.name}" renamed to "${newWalletName}"`,
            });
        } else {
            Modal.error({
                content: `Rename failed. All wallets must have a unique name.`,
            });
        }
        await updateSavedWallets(wallet);
        // Clear wallet name for form
        setNewWalletName('');
    };

    const deleteSelectedWallet = async () => {
        if (!walletDeleteValid) {
            return;
        }
        if (
            confirmationOfWalletToBeDeleted !==
            `delete ${walletToBeDeleted.name}`
        ) {
            setWalletDeleteValid(false);
            return;
        }

        // Hide modal
        setShowDeleteWalletModal(false);
        // Change wallet name
        console.log(`Deleting wallet "${walletToBeDeleted.name}"`);
        const walletDeletedSuccess = await deleteWallet(walletToBeDeleted);

        if (walletDeletedSuccess) {
            Modal.success({
                content: `Wallet "${walletToBeDeleted.name}" successfully deleted`,
            });
        } else {
            Modal.error({
                content: `Error deleting ${walletToBeDeleted.name}.`,
            });
        }
        await updateSavedWallets(wallet);
        // Clear wallet delete confirmation from form
        setConfirmationOfWalletToBeDeleted('');
    };

    const handleWalletNameInput = e => {
        const { value } = e.target;
        // validation
        if (value && value.length && value.length < 24) {
            setNewWalletNameIsValid(true);
        } else {
            setNewWalletNameIsValid(false);
        }

        setNewWalletName(value);
    };

    const handleWalletToDeleteInput = e => {
        const { value } = e.target;

        if (value && value === `delete ${walletToBeDeleted.name}`) {
            setWalletDeleteValid(true);
        } else {
            setWalletDeleteValid(false);
        }
        setConfirmationOfWalletToBeDeleted(value);
    };

    return (
        <Spin spinning={loading} indicator={CashLoadingIcon}>
            <StyledConfigure>
                {walletToBeRenamed !== null && (
                    <Modal
                        title={`Rename Wallet ${walletToBeRenamed.name}`}
                        visible={showRenameWalletModal}
                        onOk={changeWalletName}
                        onCancel={() => cancelRenameWallet()}
                    >
                        <Form style={{ width: 'auto' }}>
                            <Form.Item
                                validateStatus={
                                    newWalletNameIsValid !== null &&
                                    newWalletNameIsValid
                                        ? ''
                                        : 'error'
                                }
                                help={
                                    newWalletNameIsValid !== null &&
                                    newWalletNameIsValid
                                        ? ''
                                        : 'Wallet name must be a string between 1 and 24 characters long'
                                }
                            >
                                <Input
                                    prefix={<WalletFilled />}
                                    placeholder="Enter new wallet name"
                                    name="newName"
                                    value={newWalletName}
                                    onChange={e => handleWalletNameInput(e)}
                                />
                            </Form.Item>
                        </Form>
                    </Modal>
                )}
                {walletToBeDeleted !== null && (
                    <Modal
                        title={`Are you suer you want to delete wallet "${walletToBeDeleted.name}"?`}
                        visible={showDeleteWalletModal}
                        onOk={deleteSelectedWallet}
                        onCancel={() => cancelDeleteWallet()}
                    >
                        <Form style={{ width: 'auto' }}>
                            <Form.Item
                                validateStatus={
                                    walletDeleteValid !== null &&
                                    walletDeleteValid
                                        ? ''
                                        : 'error'
                                }
                                help={
                                    walletDeleteValid !== null &&
                                    walletDeleteValid
                                        ? ''
                                        : 'Your confirmation phrase must match exactly'
                                }
                            >
                                <Input
                                    prefix={<WalletFilled />}
                                    placeholder={`Type "delete ${walletToBeDeleted.name}" to confirm`}
                                    name="walletToBeDeletedInput"
                                    value={confirmationOfWalletToBeDeleted}
                                    onChange={e => handleWalletToDeleteInput(e)}
                                />
                            </Form.Item>
                        </Form>
                    </Modal>
                )}
                <h2>
                    <ThemedCopyOutlined /> Backup your wallet
                </h2>
                <Alert
                    style={{ marginBottom: '12px' }}
                    description="Your seed phrase is the only way to restore your wallet. Write it down. Keep it safe."
                    type="warning"
                    showIcon
                />
                {wallet && wallet.mnemonic && (
                    <StyledCollapse>
                        <Panel header="Click to reveal seed phrase" key="1">
                            <p>
                                {wallet && wallet.mnemonic
                                    ? wallet.mnemonic
                                    : ''}
                            </p>
                        </Panel>
                    </StyledCollapse>
                )}
                <StyledSpacer />
                <h2>
                    <ThemedWalletOutlined /> Manage Wallets
                </h2>
                {apiError ? (
                    <>
                        <CashLoader />
                        <p style={{ color: 'red' }}>
                            <b>An error occured on our end. Reconnecting...</b>
                        </p>
                    </>
                ) : (
                    <>
                        <PrimaryButton
                            onClick={() => updateSavedWalletsOnCreate()}
                        >
                            <PlusSquareOutlined /> New Wallet
                        </PrimaryButton>
                        <SecondaryButton
                            onClick={() => openSeedInput(!seedInput)}
                        >
                            <ImportOutlined /> Import Wallet
                        </SecondaryButton>
                        {seedInput && (
                            <>
                                <p>
                                    Copy and paste your mnemonic seed phrase
                                    below to import an existing wallet
                                </p>
                                <Form style={{ width: 'auto' }}>
                                    <Form.Item
                                        validateStatus={
                                            !formData.dirty &&
                                            !formData.mnemonic
                                                ? 'error'
                                                : ''
                                        }
                                        help={
                                            !formData.dirty &&
                                            !formData.mnemonic
                                                ? 'Mnemonic seed phrase required'
                                                : ''
                                        }
                                    >
                                        <Input
                                            prefix={<LockOutlined />}
                                            placeholder="mnemonic (seed phrase)"
                                            name="mnemonic"
                                            autoComplete="off"
                                            onChange={e => handleChange(e)}
                                            required
                                        />
                                    </Form.Item>
                                    <SmartButton
                                        disabled={!isValidMnemonic}
                                        onClick={() => submit()}
                                    >
                                        Import
                                    </SmartButton>
                                </Form>
                            </>
                        )}
                    </>
                )}
                {savedWallets && savedWallets.length > 0 && (
                    <>
                        <StyledCollapse>
                            <Panel header="Saved wallets" key="2">
                                <AWRow>
                                    <h3>{wallet.name}</h3>
                                    <h4>Currently active</h4>
                                </AWRow>
                                <div>
                                    {savedWallets.map(sw => (
                                        <SWRow key={sw.name}>
                                            <SWName>
                                                <h3>{sw.name}</h3>
                                            </SWName>

                                            <SWButtonCtn>
                                                <Edit
                                                    onClick={() =>
                                                        showPopulatedRenameWalletModal(
                                                            sw,
                                                        )
                                                    }
                                                />
                                                <Trashcan
                                                    onClick={() =>
                                                        showPopulatedDeleteWalletModal(
                                                            sw,
                                                        )
                                                    }
                                                />
                                                <button
                                                    onClick={() =>
                                                        updateSavedWalletsOnLoad(
                                                            sw,
                                                        )
                                                    }
                                                >
                                                    Activate
                                                </button>
                                            </SWButtonCtn>
                                        </SWRow>
                                    ))}
                                </div>
                            </Panel>
                        </StyledCollapse>
                    </>
                )}
                <StyledSpacer />[
                <SettingsLink
                    type="link"
                    href="https://docs.cashtabapp.com/docs/"
                    target="_blank"
                    rel="noreferrer"
                >
                    Documentation
                </SettingsLink>
                ]
            </StyledConfigure>
        </Spin>
    );
};

export default Configure;
