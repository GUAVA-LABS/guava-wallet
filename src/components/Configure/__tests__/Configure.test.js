import React from 'react';
import renderer from 'react-test-renderer';
import Configure from '../Configure';
let realUseContext;
let useContextMock;
beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});
afterEach(() => {
    React.useContext = realUseContext;
});

test('Configure without a wallet', () => {
    useContextMock.mockReturnValue({ wallet: undefined });
    const component = renderer.create(<Configure />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Configure with a wallet', () => {
    useContextMock.mockReturnValue({ wallet: { mnemonic: 'test mnemonic' } });
    const component = renderer.create(<Configure />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
