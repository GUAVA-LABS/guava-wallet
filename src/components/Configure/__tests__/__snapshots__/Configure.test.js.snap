// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure with a wallet 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div>
    <div
      className="ant-spin ant-spin-spinning"
    >
      <span
        aria-label="loading"
        className="anticon anticon-loading anticon-spin ant-spin-dot"
        role="img"
        style={
          Object {
            "color": "#ff8d00",
            "fontSize": 48,
          }
        }
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="ant-spin-container ant-spin-blur"
  >
    <div
      className="sc-jTzLTM lYoFl"
    >
      <h2>
        <span
          aria-label="copy"
          className="anticon anticon-copy sc-ifAKCX bxlMcO"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="copy"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
            />
          </svg>
        </span>
         Backup your wallet
      </h2>
      <div
        className="ant-alert ant-alert-warning ant-alert-with-description"
        data-show={true}
        role="alert"
        style={
          Object {
            "marginBottom": "12px",
          }
        }
      >
        <span
          aria-label="exclamation-circle"
          className="anticon anticon-exclamation-circle ant-alert-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="exclamation-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
            />
          </svg>
        </span>
        <div
          className="ant-alert-content"
        >
          <div
            className="ant-alert-message"
          />
          <div
            className="ant-alert-description"
          >
            Your seed phrase is the only way to restore your wallet. Write it down. Keep it safe.
          </div>
        </div>
      </div>
      <div
        className="ant-collapse ant-collapse-icon-position-left sc-bdVaJa jkIKTL"
        role={null}
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-label="right"
              className="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
            Click to reveal seed phrase
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX iUdZQY"
      />
      <h2>
        <span
          aria-label="wallet"
          className="anticon anticon-wallet sc-bZQynM cOFHGb"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="wallet"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 464H528V448h312v128zm0 264H184V184h656v200H496c-17.7 0-32 14.3-32 32v192c0 17.7 14.3 32 32 32h344v200zM580 512a40 40 0 1080 0 40 40 0 10-80 0z"
            />
          </svg>
        </span>
         Manage Wallets
      </h2>
      <button
        className="sc-bwzfXH crkPeR"
        onClick={[Function]}
      >
        <span
          aria-label="plus-square"
          className="anticon anticon-plus-square"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus-square"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
            />
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </span>
         New Wallet
      </button>
      <button
        className="sc-htpNat fajXvM"
        onClick={[Function]}
      >
        <span
          aria-label="import"
          className="anticon anticon-import"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="import"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M888.3 757.4h-53.8c-4.2 0-7.7 3.5-7.7 7.7v61.8H197.1V197.1h629.8v61.8c0 4.2 3.5 7.7 7.7 7.7h53.8c4.2 0 7.7-3.4 7.7-7.7V158.7c0-17-13.7-30.7-30.7-30.7H158.7c-17 0-30.7 13.7-30.7 30.7v706.6c0 17 13.7 30.7 30.7 30.7h706.6c17 0 30.7-13.7 30.7-30.7V765.1c0-4.3-3.5-7.7-7.7-7.7zM902 476H588v-76c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 000 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-76h314c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
         Import Wallet
      </button>
      <div
        className="sc-fjdhpX iUdZQY"
      />
      [
      <a
        className="sc-dnqmqq fnVGsZ"
        href="https://docs.cashtabapp.com/docs/"
        rel="noreferrer"
        target="_blank"
        type="link"
      >
        Documentation
      </a>
      ]
    </div>
  </div>
</div>
`;

exports[`Configure without a wallet 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div>
    <div
      className="ant-spin ant-spin-spinning"
    >
      <span
        aria-label="loading"
        className="anticon anticon-loading anticon-spin ant-spin-dot"
        role="img"
        style={
          Object {
            "color": "#ff8d00",
            "fontSize": 48,
          }
        }
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="ant-spin-container ant-spin-blur"
  >
    <div
      className="sc-jTzLTM lYoFl"
    >
      <h2>
        <span
          aria-label="copy"
          className="anticon anticon-copy sc-ifAKCX bxlMcO"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="copy"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
            />
          </svg>
        </span>
         Backup your wallet
      </h2>
      <div
        className="ant-alert ant-alert-warning ant-alert-with-description"
        data-show={true}
        role="alert"
        style={
          Object {
            "marginBottom": "12px",
          }
        }
      >
        <span
          aria-label="exclamation-circle"
          className="anticon anticon-exclamation-circle ant-alert-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="exclamation-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
            <path
              d="M464 688a48 48 0 1096 0 48 48 0 10-96 0zm24-112h48c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8z"
            />
          </svg>
        </span>
        <div
          className="ant-alert-content"
        >
          <div
            className="ant-alert-message"
          />
          <div
            className="ant-alert-description"
          >
            Your seed phrase is the only way to restore your wallet. Write it down. Keep it safe.
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX iUdZQY"
      />
      <h2>
        <span
          aria-label="wallet"
          className="anticon anticon-wallet sc-bZQynM cOFHGb"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="wallet"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 464H528V448h312v128zm0 264H184V184h656v200H496c-17.7 0-32 14.3-32 32v192c0 17.7 14.3 32 32 32h344v200zM580 512a40 40 0 1080 0 40 40 0 10-80 0z"
            />
          </svg>
        </span>
         Manage Wallets
      </h2>
      <button
        className="sc-bwzfXH crkPeR"
        onClick={[Function]}
      >
        <span
          aria-label="plus-square"
          className="anticon anticon-plus-square"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus-square"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
            />
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </span>
         New Wallet
      </button>
      <button
        className="sc-htpNat fajXvM"
        onClick={[Function]}
      >
        <span
          aria-label="import"
          className="anticon anticon-import"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="import"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M888.3 757.4h-53.8c-4.2 0-7.7 3.5-7.7 7.7v61.8H197.1V197.1h629.8v61.8c0 4.2 3.5 7.7 7.7 7.7h53.8c4.2 0 7.7-3.4 7.7-7.7V158.7c0-17-13.7-30.7-30.7-30.7H158.7c-17 0-30.7 13.7-30.7 30.7v706.6c0 17 13.7 30.7 30.7 30.7h706.6c17 0 30.7-13.7 30.7-30.7V765.1c0-4.3-3.5-7.7-7.7-7.7zM902 476H588v-76c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 000 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-76h314c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
         Import Wallet
      </button>
      <div
        className="sc-fjdhpX iUdZQY"
      />
      [
      <a
        className="sc-dnqmqq fnVGsZ"
        href="https://docs.cashtabapp.com/docs/"
        rel="noreferrer"
        target="_blank"
        type="link"
      >
        Documentation
      </a>
      ]
    </div>
  </div>
</div>
`;
